src/humanloop/otel/exporter.py

```diff
@@ -0,0 +1,305 @@
+        """
+        super().__init__()
+        self._client = client
+        # Uploaded spans translate to a Log on Humanloop. The IDs are required to link Logs in a Flow Trace
+        self._span_id_to_uploaded_log_id: dict[int, Optional[str]] = {}
+        # Work queue for the threads uploading the spans
+        self._upload_queue: Queue = Queue()
+        # Worker threads to export the spans
+        self._threads: list[Thread] = [
+            Thread(
+                target=self._do_work,
+                daemon=True,
+            )
+            for _ in range(worker_threads or self.DEFAULT_NUMBER_THREADS)
+        ]
+        # Signals threads no more work will arrive and
+        # they should wind down if the queue is empty
+        self._shutdown: bool = False
+        for thread in self._threads:
+            thread.start()
+
+    def export(self, spans: trace.Sequence[ReadableSpan]) -> SpanExportResult:
+        def is_evaluated_file(
+            span: ReadableSpan,
+            evaluation_context: Optional[EvaluationContext],
+        ) -> bool:
+            if evaluation_context is None:
+                return False
+
+            return span.attributes.get(HUMANLOOP_PATH_KEY) == evaluation_context["path"]  # type: ignore
+
+        if not self._shutdown:
+            try:
+                evaluation_context = self._client.evaluation_context_variable.get()
+                if len(spans) > 1:
+                    raise RuntimeError("HumanloopSpanExporter expected a single span when running an evaluation")
+                if not is_evaluated_file(spans[0], evaluation_context):
+                    evaluation_context = None
+            except LookupError:
+                evaluation_context = None
+            for span in spans:
+                if is_humanloop_span(span):
+                    # We pass the EvaluationContext from the eval_run utility thread to
+                    # the export thread so the .log action works as expected
+                    self._upload_queue.put(
+                        (
+                            span,
+                            copy.deepcopy(evaluation_context),
+                        )
+                    )
+            # Reset the EvaluationContext so run eval does not
+            # create a duplicate Log
+            if evaluation_context is not None and is_evaluated_file(
+                spans[0],
+                evaluation_context,
+            ):
+                # Mark the EvaluationContext as used
+                self._client.evaluation_context_variable.set(None)
+            return SpanExportResult.SUCCESS
+        else:
+            logger.warning("HumanloopSpanExporter is shutting down, not accepting new spans")
+            return SpanExportResult.FAILURE
+
+    def shutdown(self) -> None:
+        self._shutdown = True
+        for thread in self._threads:
+            thread.join()
+
+    def force_flush(self, timeout_millis: int = 3000) -> bool:
```

Comment:
find `millis` slightly weird.
if we're providing units, why not use the full name?
https://docs.python.org/3/library/datetime.html#timedelta-objects
`timedelta`s use "milliseconds" for example.

Reply:
it's a method inherited from OTel, i don't like it either